bundles/org.openhab.binding.boschshc/src/main/java/org/openhab/binding/boschshc/internal/devices/bridge/LongPolling.java

```diff
@@ -144,65 +161,73 @@ private void longPoll(BoschHttpClient httpClient, String subscriptionId) {
         request.send(new BufferingResponseListener() {
             @Override
             public void onComplete(@Nullable Result result) {
-                Throwable failure = result != null ? result.getFailure() : null;
-                if (failure != null) {
-                    if (failure instanceof ExecutionException) {
-                        if (failure.getCause() instanceof AbortLongPolling) {
-                            logger.debug("Canceling long polling for subscription id {} because it was aborted",
-                                    subscriptionId);
-                        } else {
-                            longPolling.handleFailure.accept(new LongPollingFailedException(
-                                    "Unexpected exception during long polling request", failure));
-                        }
-                    } else {
-                        longPolling.handleFailure.accept(new LongPollingFailedException(
-                                "Unexpected exception during long polling request", failure));
-                    }
-                } else {
-                    longPolling.onLongPollResponse(httpClient, subscriptionId, this.getContentAsString());
-                }
+                // NOTE: This handler runs inside the HTTP thread, so we schedule the response handling in a new thread
+                // because the HTTP thread is terminated after the timeout expires.
+                scheduler.execute(() -> longPolling.onLongPollComplete(httpClient, subscriptionId, result,
+                        this.getContentAsString()));
             }
         });
     }
 
-    private void onLongPollResponse(BoschHttpClient httpClient, String subscriptionId, String content) {
+    /**
+     * This is the handler for responses of long poll requests.
+     * 
+     * @param httpClient HTTP client which received the response
+     * @param subscriptionId Id of subscription the response is for
+     * @param result Complete result of the response
+     * @param content Content of the response
+     */
+    private void onLongPollComplete(BoschHttpClient httpClient, String subscriptionId, @Nullable Result result,
+            String content) {
         // Check if thing is still online
         if (this.aborted) {
             logger.debug("Canceling long polling for subscription id {} because it was aborted", subscriptionId);
             return;
         }
 
-        logger.debug("Long poll response: {}", content);
+        Throwable failure = result != null ? result.getFailure() : null;
+        if (failure != null) {
```

Comment:
New if/else code is quite long.
I suggest to extract to methods for the two blocks and result in e.g.

        if (failure != null) {
            throwExceptionForFailure(failure);
        } else {
            processLongPollContent(...);
        }

Reply:
I agree that it got a bit long, but I like to keep it in one place for now to follow the logic better in one place. Almost half of the lines is logging anyway (which we probably could remove when the binding is stable). Is that okay for you?

Reply:
OK